bundle/edu.gemini.phase2.skeleton.servlet/src/main/scala/edu/gemini/phase2/skeleton/factory/SpProgramFactory.scala

```diff
@@ -201,11 +201,16 @@ object SpProgramFactory {
       timeAccountingRatios(proposal) match {
         case Nil    => None
         case ratios =>
+          val progTime = hoursFromObservations(proposal, _.progTime)
+          val partTime = hoursFromObservations(proposal, _.partTime)
+          require((hrs - (progTime |+| partTime).hours).abs < 1e-1,
+            "Awarded time must be equal to the sum of program time and partner calibration time")
```

Comment:
Agree that this feels a bit risky. This should be enforced somehow, and if it currently fails, the ODB will log an exception and the skeleton script will return it.

My concern is accumulated floating point arithmetic errors across many observations. I'm not sure if this will ever happen.

We should somehow, though, ensure that the ITAC awarded time is equal to the P1 required time here.